src/demo/demo.js

```diff
@@ -101,3 +55,80 @@ const getToken = function(onSuccess) {
   }
   request.send()
 }
+
+
+class SDK extends Component{
+  componentDidMount () {
+    this.initSDK(this.props.options)
+  }
+
+  componentWillReceiveProps({options}){
+    if (this.state.onfidoSdk){
+      this.state.onfidoSdk.setOptions(options)
+    }
+  }
+
+  componentWillUnmount () {
+    if (this.state.onfidoSdk) this.state.onfidoSdk.tearDown()
+  }
+
+  initSDK = (options)=> {
+    const onfidoSdk = Onfido.init(options)
+    this.setState({onfidoSdk})
+  }
+
+  shouldComponentUpdate () {
+    return false
+  }
+
+  render () {
+    return <div id="onfido-mount"></div>
+  }
+}
+
+class Demo extends Component{
+  constructor (props) {
+    super(props)
+    getToken((token)=> {
+      this.setState({token})
+    })
+  }
+
+  state = {
+    token: null,
+    isModalOpen: false
+  }
+
+  sdkOptions = ()=> ({
+    token: this.state.token,
+    useModal,
+    onComplete: () => {
+      /*callback for when */ console.log("everything is complete")
+    },
+    isModalOpen: this.state.isModalOpen,
+    language,
+    steps,
+    mobileFlow: !!options.link_id,
+    onModalRequestClose: () => {
+      this.setState({isModalOpen: false})
+    }
+  })
+
+  render = () => (
+    <div class="container">
+      { useModal &&
+        <button
+          id="button"
+          onClick={ () => this.setState({isModalOpen: true})}>
+            Verify identity
+        </button>
+      }
+      {options.async === "false" && this.state.token === null?
+        <span/> : <SDK options={this.sdkOptions()}></SDK>
```

Comment:
[minor] why `<span/>` rather than `null`?

Reply:
no reason, I will change it